compute_endpoint/tests/unit/test_endpoint_unit.py

```diff
@@ -831,3 +849,126 @@ def test_update_config_file_retains_order(fs):
 
     target_config = target_path.read_text()
     assert original_config == target_config
+
+
+@pytest.mark.parametrize(
+    "config_keys",
+    (
+        (
+            "admins",
+            "display_name",
+            "allowed_functions",
+            "authentication_policy",
+            "subscription_id",
+            "debug",
+            "amqp_port",
+            "heartbeat_period",
+        ),
+        (
+            "admins",
+            "display_name",
+            "allowed_functions",
+            "authentication_policy",
+            "subscription_id",
+        ),
+        ("foo", "bar", "baz"),
+        ("debug", "amqp_port", "heartbeat_period"),
+        ("admins", "foo", "debug", "bar"),
+    ),
+)
+def test_migrate_to_template_capable_success(
+    fs, mock_ep_data, mock_get_config, config_keys
+):
+    ep, ep_dir, *_ = mock_ep_data
+    config_dict = {k: "some value" for k in config_keys}
+    mock_get_config.return_value = SimpleNamespace(
+        source_content=yaml.dump(config_dict)
+    )
+    ep._config_file_path(ep_dir).write_text(yaml.safe_dump(config_dict))
+
+    ep.migrate_to_template_capable(ep_dir)
+
+    assert ep._config_file_path(ep_dir).exists()
+    assert ep.user_config_template_path(ep_dir).exists()
+
+    new_conf = yaml.safe_load(ep._config_file_path(ep_dir).open())
+    for k in new_conf:
+        assert k in config_dict
+
+    new_templ = yaml.safe_load(ep.user_config_template_path(ep_dir).open())
+    for k in new_templ:
+        assert k in config_dict
+
+    config_only_keys = {
+        "admins",
+        "display_name",
+        "allowed_functions",
+        "authentication_policy",
+        "subscription_id",
+    }
+
+    shared_keys = {
+        "debug",
+        "amqp_port",
+        "heartbeat_period",
+    }
+
+    for k in config_dict:
+        assert k in new_conf or k in new_templ
```

Comment:
At least one assert making sure the value carries over?  e.g. ``assert ..... == "some value"``